Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -577,22 +595,32 @@ public Vector3 GetPointerGrabPoint(uint pointerId)
         public virtual void OnPointerDown(MixedRealityPointerEventData eventData)
         {
             if (eventData.used ||
-                    eventData.Pointer == null ||
-                    eventData.Pointer.Result == null ||
-                    (!allowFarManipulation && eventData.Pointer as IMixedRealityNearPointer == null))
+                eventData.Pointer == null ||
+                eventData.Pointer.Result == null ||
+                (!allowFarManipulation && eventData.Pointer as IMixedRealityNearPointer == null))
             {
                 return;
             }
 
             // If we only allow one handed manipulations, check there is no hand interacting yet.
-            if (manipulationType != ManipulationHandFlags.OneHanded || pointerIdToPointerMap.Count == 0)
+            if (manipulationType != ManipulationHandFlags.OneHanded || pointerDataList.Count == 0)
             {
                 uint id = eventData.Pointer.PointerId;
-                // Ignore poke pointer events
-                if (!pointerIdToPointerMap.ContainsKey(id))
+                bool pointerPresent = false;
+                int pointerDataListCount = pointerDataList.Count;
+                for (int i = 0; i < pointerDataListCount; i++)
```

Comment:
Maybe all of these for loops can be consolidated to a helper function rather than many for loops.

Reply:
Consolidated! These three, at least. The rest didn't follow a reasonably similar enough pattern.